ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -43,20 +43,20 @@
     #where=np_frontend_helpers.where(),
 
 )
-def test_numpy_diagonal(
+def test_diagonal(
```

Comment:
Leave that as `test_numpy_diagonal`. We actually mention the frontend in the test function's name. Don't know why 😅

Reply:
> Leave that as `test_numpy_diagonal`. We actually mention the frontend in the test function's name. Don't know why 😅

i just changed the name of the function to numpy_diagonal and it works (before seeing your comments, :). Now the key error is removed. It's not working with numpy_diagonal, i checked again.